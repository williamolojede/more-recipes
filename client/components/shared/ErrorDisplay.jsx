import React from 'react';
import { errorDisplayPropTypes } from '../../config/proptypes';

const ErrorDisplay = props => (
  <div className="error z-depth-1">
    <span className="error__cancel">X</span>
    <p className="error__message">{ props.message }</p>
  </div>
);

ErrorDisplay.propTypes = errorDisplayPropTypes;

export default ErrorDisplay;
