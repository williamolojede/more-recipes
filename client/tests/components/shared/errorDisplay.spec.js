import React from 'react';
import ErrorDisplay from '../../../components/shared/ErrorDisplay';

function setup() {
  const wrapper = shallow(<ErrorDisplay message="shit went wrong" />);

  return {
    wrapper
  };
}

describe('Error Display', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
