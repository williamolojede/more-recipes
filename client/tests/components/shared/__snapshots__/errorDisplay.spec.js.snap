// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Display should render with right amount of elements 1`] = `
<div
  className="error z-depth-1"
>
  <span
    className="error__cancel"
  >
    X
  </span>
  <p
    className="error__message"
  >
    shit went wrong
  </p>
</div>
`;
